benchmark/parquet/large_schema_union_by_name.benchmark

```diff
@@ -0,0 +1,40 @@
+# name: benchmark/parquet/large_schema_union_by_name.benchmark
```

Comment:
Can we add a test that tests the new parameter for CSV as well?

Reply:
I got this PR working for CSVs as well. For testing, I added an existing `union_by_name` benchmark to the regression test suite and several tests for `cache_union_readers=False`.